internal/vendors/service.go

```diff
@@ -40,11 +48,52 @@ func (v *VendorService) GetLocations(ctx context.Context) ([]string, error) {
 	return v.vendorDBAccessor.GetAllLocations(ctx)
 }
 
+func (v *VendorService) BlastEmail(ctx context.Context, vendorIDs []string, template emailTemplate) error {
+	vendors, err := v.vendorDBAccessor.BulkGetByIDs(ctx, vendorIDs)
+	if err != nil {
+		return err
+	}
+
+	errCh := make(chan error, len(vendors))
+	defer close(errCh)
+
+	var wg sync.WaitGroup
+
+	for _, vendor := range vendors {
+		wg.Add(1)
```

Comment:
I think we should limit the number of worker to prevent system overload (even though goroutines is already lightweight). We can control the concurrency using Worker Pool and Semaphore here.

Reply:
there's no need for semaphore since I don't think we have any critical sections here (?)

Reply:
also worker pool wise, I think we'll just be handling around 50 email blasts at once, the pool would simply be useless on such a small scale

Reply:
the semaphore isn't being used for thread safety or to protect critical sections but for concurrency limiter to control the number of goroutines running at the same time